enterprise/tailnet/pgcoord.go

```diff
@@ -454,6 +474,9 @@ func newBinder(ctx context.Context,
 		workQ:         newWorkQ[bKey](ctx),
 	}
 	go b.handleBindings()
+	// add to the waitgroup immediately to avoid any races waiting for it before
+	// the workers start.
+	b.workerWG.Add(numBinderWorkers)
```

Comment:
Is there a chance that `<-startWorkers` below (i.e. `fHB`) doesn't get closed (e.g. some error during startup), and thus, these waitgroups never resolving?

(I didn't try to dig in as to how or where `fHB` is closed as it's not obvious from this PR.)

Reply:
It gets closed unconditionally after we send the first heartbeat (success or fail).